### DISCUSSION TITLE
remove all subtitles exept ar ?

### DISCUSSION THREAD

**Discussion by Aabayoumy** [ptr727/PlexCleaner#347] [Q&A] [answered]
Hi
I set this setting  "KeepLanguages": ["ar"],
did that related to subtitles or audio track?

**Comment by Aabayoumy** [comment]
i test it and removed all subtitles 
16:20:23 [INF] <1> "Remove" : Type: "SubtitleInfo", Format: "SubRip/SRT", Codec: "S_TEXT/UTF8", Language: "eng", LanguageIetf: "en", Id: 2, Number: 3, Title: "GalaxyTV - small excellence!", Flags: Default, State: Remove, HasErrors: True, HasTags: True
16:20:23 [INF] <1> "Remove" : Type: "SubtitleInfo", Format: "SubRip/SRT", Codec: "S_TEXT/UTF8", Language: "eng", LanguageIetf: "en", Id: 3, Number: 4, Title: "English [SDH]", Flags: HearingImpaired, State: Remove, HasErrors: True, HasTags: False
16:20:23 [INF] <1> "Remove" : Type: "SubtitleInfo", Format: "SubRip/SRT", Codec: "S_TEXT/UTF8", Language: "ara", LanguageIetf: "ar", Id: 4, Number: 5, Title: "Arabic", Flags: None, State: Remove, HasErrors: True, HasTags: True
16:20:23 [INF] <1> "Remove" : Type: "SubtitleInfo", Format: "SubRip/SRT", Codec: "S_TEXT/UTF8", Language: "baq", LanguageIetf: "eu", Id: 5, Number: 6, Title: "Basque", Flags: None, State: Remove, HasErrors: True, HasTags: True
16:20:23 [INF] <1> "Remove" : Type: "SubtitleInfo", Format: "SubRip/SRT", Codec: "S_TEXT/UTF8", Language: "cat", LanguageIetf: "ca", Id: 6, Number: 7, Title: "Catalan", Flags: None, State: Remove, HasErrors: True, HasTags: True
16:20:23 [INF] <1> "Remove" : Type: "SubtitleInfo", Format: "SubRip/SRT", Codec: "S_TEXT/UTF8", Language: "chi", LanguageIetf: "zh", Id: 7, Number: 8, Title: "Chinese Simplified", Flags: None, State: Remove, HasErrors: True, HasTags: True

even try with 
 "KeepLanguages": ["ar", "ara"],

**Reply by ptr727** [reply]
Applies to all track types, attach full config and attach full log output.

**Reply by Aabayoumy** [reply]
[PlexCleaner.json](https://github.com/ptr727/PlexCleaner/files/14921143/PlexCleaner.json)
[PlexCleaner-3body.log](https://github.com/ptr727/PlexCleaner/files/14921163/PlexCleaner-3body.log)

**Reply by Aabayoumy** [reply]
log for other file
[PlexCleaner-3body.log](https://github.com/ptr727/PlexCleaner/files/14921260/PlexCleaner-3body.log)

**Reply by ptr727** [reply]
Both log files are for the `removesubtitles` command so "all" subtitles will be removed.

Per the docs:

```
removesubtitles:
Remove all subtitle tracks from the media files.
Useful when subtitles are forced and contains offensive language or advertising.
```

If you want normal logic you need to use the `process` or `monitor` command, and you need to change `"RemoveUnwantedLanguageTracks": false,` to `true` in settings so `KeepLanguages` setting will be used.

**Reply by Aabayoumy** [reply]
i did that thx, but now not remove any thing
[PlexCleaner.json](https://github.com/ptr727/PlexCleaner/files/14922514/PlexCleaner.json)
[PlexCleaner-3body.log](https://github.com/ptr727/PlexCleaner/files/14922515/PlexCleaner-3body.log)

**Reply by ptr727** [reply]
Looks like your ffmpeg settings makes ffmpeg unhappy, log shows failures before even getting to processing:

```
2024-04-09 19:54:48.808 +02:00 [INF] <1> Executing FfMpeg : "-analyzeduration 2147483647 -probesize 2147483647 -hwaccel cuda -hwaccel_output_format cuda -i \"P:\TVSHOW\3 Body Problem\Season 01\3 Body Problem - S01E08 - Wallfacer WEBRip-720p GalaxyTV.mkv\" -max_muxing_queue_size 1024 -abort_on empty_output -hide_banner -nostats -filter:v idet -an -f rawvideo -y NUL"
2024-04-09 19:54:49.695 +02:00 [ERR] <1> Failed to count interlaced frames : "3 Body Problem - S01E08 - Wallfacer WEBRip-720p GalaxyTV.mkv"
2024-04-09 19:54:49.696 +02:00 [ERR] <1> "Process" Error : "P:\TVSHOW\3 Body Problem\Season 01\3 Body Problem - S01E08 - Wallfacer WEBRip-720p GalaxyTV.mkv"
```

Try to execute ffmpeg using the commands as reported in the log output and see what it complains about?

`ffmpeg -analyzeduration 2147483647 -probesize 2147483647 -hwaccel cuda -hwaccel_output_format cuda -i \"P:\TVSHOW\3 Body Problem\Season 01\3 Body Problem - S01E08 - Wallfacer WEBRip-720p GalaxyTV.mkv\" -max_muxing_queue_size 1024 -abort_on empty_output -hide_banner -nostats -filter:v idet -an -f rawvideo -y NUL`

**Reply by Aabayoumy** [reply]
.\ffmpeg -analyzeduration 2147483647 -probesize 2147483647 -hwaccel cuda -hwaccel_output_format cuda -i \"P:\TVSHOW\3 Body Problem\Season 01\3 Body Problem - S01E08 - Wallfacer WEBRip-720p GalaxyTV.mkv\" -max_muxing_queue_size 1024 -abort_on empty_output -hide_banner -nostats -filter:v idet -an -f rawvideo -y NUL
ffmpeg version 7.0-full_build-www.gyan.dev Copyright (c) 2000-2024 the FFmpeg developers
  built with gcc 13.2.0 (Rev5, Built by MSYS2 project)
  configuration: --enable-gpl --enable-version3 --enable-static --disable-w32threads --disable-autodetect --enable-fontconfig --enable-iconv --enable-gnutls --enable-libxml2 --enable-gmp --enable-bzlib --enable-lzma --enable-libsnappy --enable-zlib --enable-librist --enable-libsrt --enable-libssh --enable-libzmq --enable-avisynth --enable-libbluray --enable-libcaca --enable-sdl2 --enable-libaribb24 --enable-libaribcaption --enable-libdav1d --enable-libdavs2 --enable-libuavs3d --enable-libxevd --enable-libzvbi --enable-librav1e --enable-libsvtav1 --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxavs2 --enable-libxeve --enable-libxvid --enable-libaom --enable-libjxl --enable-libopenjpeg --enable-libvpx --enable-mediafoundation --enable-libass --enable-frei0r --enable-libfreetype --enable-libfribidi --enable-libharfbuzz --enable-liblensfun --enable-libvidstab --enable-libvmaf --enable-libzimg --enable-amf --enable-cuda-llvm --enable-cuvid --enable-dxva2 --enable-d3d11va --enable-d3d12va --enable-ffnvcodec --enable-libvpl --enable-nvdec --enable-nvenc --enable-vaapi --enable-libshaderc --enable-vulkan --enable-libplacebo --enable-opencl --enable-libcdio --enable-libgme --enable-libmodplug --enable-libopenmpt --enable-libopencore-amrwb --enable-libmp3lame --enable-libshine --enable-libtheora --enable-libtwolame --enable-libvo-amrwbenc --enable-libcodec2 --enable-libilbc --enable-libgsm --enable-libopencore-amrnb --enable-libopus --enable-libspeex --enable-libvorbis --enable-ladspa --enable-libbs2b --enable-libflite --enable-libmysofa --enable-librubberband --enable-libsoxr --enable-chromaprint
  libavutil      59.  8.100 / 59.  8.100
  libavcodec     61.  3.100 / 61.  3.100
  libavformat    61.  1.100 / 61.  1.100
  libavdevice    61.  1.100 / 61.  1.100
  libavfilter    10.  1.100 / 10.  1.100
  libswscale      8.  1.100 /  8.  1.100
  libswresample   5.  1.100 /  5.  1.100
  libpostproc    58.  1.100 / 58.  1.100
[in#0 @ 0000024ca29da780] Error opening input: Invalid argument
Error opening input file \P:\TVSHOW\3 Body Problem\Season 01\3 Body Problem - S01E08 - Wallfacer WEBRip-720p GalaxyTV.mkv" -max_muxing_queue_size 1024 -abort_on empty_output -hide_banner -nostats -filter:v idet -an -f rawvideo -y NUL.
Error opening input files: Invalid argument

**Reply by Aabayoumy** [reply]
.\ffmpeg.exe
ffmpeg version 7.0-full_build-www.gyan.dev Copyright (c) 2000-2024 the FFmpeg developers
  built with gcc 13.2.0 (Rev5, Built by MSYS2 project)
  configuration: --enable-gpl --enable-version3 --enable-static --disable-w32threads --disable-autodetect --enable-fontconfig --enable-iconv --enable-gnutls --enable-libxml2 --enable-gmp --enable-bzlib --enable-lzma --enable-libsnappy --enable-zlib --enable-librist --enable-libsrt --enable-libssh --enable-libzmq --enable-avisynth --enable-libbluray --enable-libcaca --enable-sdl2 --enable-libaribb24 --enable-libaribcaption --enable-libdav1d --enable-libdavs2 --enable-libuavs3d --enable-libxevd --enable-libzvbi --enable-librav1e --enable-libsvtav1 --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxavs2 --enable-libxeve --enable-libxvid --enable-libaom --enable-libjxl --enable-libopenjpeg --enable-libvpx --enable-mediafoundation --enable-libass --enable-frei0r --enable-libfreetype --enable-libfribidi --enable-libharfbuzz --enable-liblensfun --enable-libvidstab --enable-libvmaf --enable-libzimg --enable-amf --enable-cuda-llvm --enable-cuvid --enable-dxva2 --enable-d3d11va --enable-d3d12va --enable-ffnvcodec --enable-libvpl --enable-nvdec --enable-nvenc --enable-vaapi --enable-libshaderc --enable-vulkan --enable-libplacebo --enable-opencl --enable-libcdio --enable-libgme --enable-libmodplug --enable-libopenmpt --enable-libopencore-amrwb --enable-libmp3lame --enable-libshine --enable-libtheora --enable-libtwolame --enable-libvo-amrwbenc --enable-libcodec2 --enable-libilbc --enable-libgsm --enable-libopencore-amrnb --enable-libopus --enable-libspeex --enable-libvorbis --enable-ladspa --enable-libbs2b --enable-libflite --enable-libmysofa --enable-librubberband --enable-libsoxr --enable-chromaprint
  libavutil      59.  8.100 / 59.  8.100
  libavcodec     61.  3.100 / 61.  3.100
  libavformat    61.  1.100 / 61.  1.100
  libavdevice    61.  1.100 / 61.  1.100
  libavfilter    10.  1.100 / 10.  1.100
  libswscale      8.  1.100 /  8.  1.100
  libswresample   5.  1.100 /  5.  1.100
  libpostproc    58.  1.100 / 58.  1.100
Universal media converter
usage: ffmpeg [options] [[infile options] -i infile]... {[outfile options] outfile}...

**Comment by Aabayoumy** [comment]
maybe issue with ver 7 ?

**Reply by ptr727** [reply]
I have tested with v7 and did not see such an issue.

For your test command it looks more like the path you are trying is not valid: `Error opening input file \P:\TVSHOW\3 Body Problem\Season 01\3 Body Problem - S01E08 - Wallfacer WEBRip-720p GalaxyTV.mkv"`, should start with a `"P:\` not `\P:\`? The path in the command should be properly quoted if it contains space, maybe try again and pay attention to quoted paths with spaces.

From your settings you have also modified the defaults to add hardware acceleration for cuda, but I don't see it in the test output, I'd suggest you try first with default settings, then when you know things work, you start fiddling with adding hw acceleration tweaks.

**Reply by Aabayoumy** [reply]
YEs, i resolve path issue now 
PS C:\PlexCleaner\Tools\FfMpeg\bin>  .\ffmpeg -analyzeduration 2147483647 -probesize 2147483647 -hwaccel cuda -hwaccel_output_format cuda -i "P:\TVSHOW\3 Body Problem\Season 01\3 Body Problem - S01E08 - Wallfacer WEBRip-720p GalaxyTV.mkv" -max_muxing_queue_size 1024 -abort_on empty_output -hide_banner -nostats -filter:v idet -an -f rawvideo -y NUL
Input #0, matroska,webm, from 'P:\TVSHOW\3 Body Problem\Season 01\3 Body Problem - S01E08 - Wallfacer WEBRip-720p GalaxyTV.mkv':
  Metadata:
    encoder         : libebml v1.4.5 + libmatroska v1.7.1
    creation_time   : 2024-04-10T14:09:27.000000Z
  Duration: 00:56:42.15, start: 0.000000, bitrate: 540 kb/s
  Stream #0:0(eng): Video: h264 (High), yuv420p(tv, progressive), 1280x720 [SAR 1:1 DAR 16:9], 23.98 fps, 23.98 tbr, 1k tbn (default)
  Stream #0:1(eng): Audio: aac (LC), 48000 Hz, stereo, fltp (default)
  Stream #0:2(eng): Subtitle: subrip (srt) (default)
  Stream #0:3(eng): Subtitle: subrip (srt) (hearing impaired)
      Metadata:
        title           : English [SDH]
  Stream #0:4(ara): Subtitle: subrip (srt)
[Parsed_idet_0 @ 0000026a6fac5840] Repeated Fields: Neither:     0 Top:     0 Bottom:     0
[Parsed_idet_0 @ 0000026a6fac5840] Single frame detection: TFF:     0 BFF:     0 Progressive:     0 Undetermined:     0
[Parsed_idet_0 @ 0000026a6fac5840] Multi frame detection: TFF:     0 BFF:     0 Progressive:     0 Undetermined:     0
Stream mapping:
  Stream #0:0 -> #0:0 (h264 (native) -> rawvideo (native))
Press [q] to stop, [?] for help
Impossible to convert between the formats supported by the filter 'graph -1 input from stream 0:0' and the filter 'auto_scale_0'
[Parsed_idet_0 @ 0000026a6fac6a40] Repeated Fields: Neither:     0 Top:     0 Bottom:     0
[Parsed_idet_0 @ 0000026a6fac6a40] Single frame detection: TFF:     0 BFF:     0 Progressive:     0 Undetermined:     0
[Parsed_idet_0 @ 0000026a6fac6a40] Multi frame detection: TFF:     0 BFF:     0 Progressive:     0 Undetermined:     0
[vf#0:0 @ 0000026a6fa31d80] Error reinitializing filters!
[vf#0:0 @ 0000026a6fa31d80] Task finished with error code: -40 (Function not implemented)
[vf#0:0 @ 0000026a6fa31d80] Terminating thread with return code -40 (Function not implemented)
[vost#0:0/rawvideo @ 0000026a6fa7b2c0] Could not open encoder before EOF
[vost#0:0/rawvideo @ 0000026a6fa7b2c0] Task finished with error code: -22 (Invalid argument)
[vost#0:0/rawvideo @ 0000026a6fa7b2c0] Terminating thread with return code -22 (Invalid argument)
[out#0/rawvideo @ 0000026a6fa56340] Nothing was written into output file, because at least one of its streams received no packets.
frame=    0 fps=0.0 q=0.0 Lsize=       0KiB time=N/A bitrate=N/A speed=N/A
Conversion failed!

**Reply by ptr727** [reply]
Try without `-hwaccel cuda -hwaccel_output_format cuda`?

**Reply by ptr727** [reply, marked as answer]
You could also try the [develop](https://github.com/ptr727/PlexCleaner/tree/develop) pre-release version, I added logging for `idet` failures and now only use the FfMpeg's global convert option for encoding, not for any reading or muxing operations only. As I could not repro your issue, it may not actually fix it, but seems to be a good preventative change.
